File: src/main/java/net/shotbow/interestingfish/utility/ItemUtility.java
package net.shotbow.interestingfish.utility;

import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.PlayerInventory;
import org.bukkit.inventory.meta.ItemMeta;

import java.util.List;

/**
 * User: Austin
 * Date: 2/22/13
 * Time: 9:44 PM
 * (c) lazertester
 */
public class ItemUtility {

    public static void renameItem(ItemStack item, String newName) {
        final ItemMeta itemMeta = item.getItemMeta();
        itemMeta.setDisplayName(ChatColor.RESET + newName);
        item.setItemMeta(itemMeta);
    }

    public static void setLore(ItemStack item, List<String> lore) {
        ItemMeta meta = item.getItemMeta();
        meta.setLore(lore);
        item.setItemMeta(meta);
    }

    public static boolean fishingRodHasLuck(PlayerInventory inventory) {
        ItemStack item = inventory.getItemInMainHand();
        if (item.getType() != Material.FISHING_ROD) {
            item = inventory.getItemInOffHand();
        }
        if (item.getType() == Material.FISHING_ROD) {
            if (item.hasItemMeta()) {
                return item.getItemMeta().hasEnchant(Enchantment.LUCK);
            }
        }
        return false;
    }
}

File: src/main/kotlin/net/shotbow/interestingfish/util/ItemUtility.kt
package net.shotbow.interestingfish.util

import org.bukkit.Material
import org.bukkit.Tag
import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.PlayerInventory
import org.bukkit.inventory.meta.ItemMeta

object ItemUtility {
    fun ItemStack.meta(updates: ItemMeta.() -> Unit) {
        itemMeta = itemMeta.also(updates)
    }

    fun ItemStack.isFish() = Tag.ITEMS_FISHES.isTagged(type)

    fun fishingRodHasLuck(inventory: PlayerInventory): Boolean =
        inventory.itemInEitherHand(Material.FISHING_ROD)
            ?.let { it.hasItemMeta() && it.itemMeta.hasEnchant(Enchantment.LUCK) } ?: false

    private fun PlayerInventory.itemInEitherHand(material: Material) =
        itemInMainHand.takeIf { it.type == material } ?: itemInOffHand.takeIf { it.type == material }
}
